web_programming/mattermost_message.py

```diff
@@ -0,0 +1,25 @@
+import os
+import requests
+
+
+WEBHOOK_URL = os.getenv("YOUR_WEBHOOK_URL", "")
+
+
+def send_mattermost_message(
+    message: str,
+    username: str,
+    webhook_url: str,
+) -> None:
+    payload = {
+        "text": message,
+        "username": username
+    }
+    response = requests.post(webhook_url, json=payload)
+    if response.status_code != 200:
+        raise Exception(
+            "Failed to send a message in Mattermost: %s" % response.text,
```

Comment:
@dhruvmanila Does this advise conflict with https://beta.ruff.rs/docs/rules/#flake8-errmsg-em ?

Reply:
No, the rule suggests to use a variable for the message and then use the variable in the exception:
```python
msg = f"error message: {foo}"
raise Execption(msg)

# instead of
raise Exception(f"error messsage: {foo}")
```